common/src/main/java/com/redhat/cloud/notifications/models/validation/AllowedHttpsSchemeValidator.java

```diff
@@ -0,0 +1,39 @@
+package com.redhat.cloud.notifications.models.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+public class AllowedHttpsSchemeValidator implements ConstraintValidator<AllowedHttpsScheme, String> {
+
+    private static final String SCHEME_HTTP = "http";
+    private static final String SCHEME_HTTPS = "https";
+
+    @Override
+    public void initialize(final AllowedHttpsScheme constraintAnnotation) {
+        ConstraintValidator.super.initialize(constraintAnnotation);
+    }
+
+    /**
+     * Validates that the provided raw URL contains an allowed HTTP scheme.
+     * @param rawUrl the {@link String} URL to validate.
+     * @param constraintValidatorContext the constraint validation context.
+     * @return true if the {@link String} contains an allowed HTTP scheme.
+     */
+    @Override
+    public boolean isValid(final String rawUrl, final ConstraintValidatorContext constraintValidatorContext) {
+        try {
+            final var url = new URL(rawUrl);
+            final URI uri = url.toURI();
+
+            final String scheme = uri.getScheme();
+
+            return SCHEME_HTTP.equals(scheme) || SCHEME_HTTPS.equals(scheme);
+        } catch (MalformedURLException | URISyntaxException e) {
```

Comment:
Will this constraint validator always be executed after `WellFormedUrlValidator`? Otherwise, this validator may return a validator error saying that the HTTP scheme is invalid while the actual problem is a malformed URL.

Reply:
We don't know. This is one of the reasons why I initially wrote the three annotations all in one, though that came with the drawback of not having a precise enough message.

I could probably create a validation group which enforces the order of these three validations, and then apply that in the corresponding "endpoint resource".

Reply:
`MalformedURLException` is caught by the 3 validators which means that a malformed URL will trigger a validation error with any of the 3 validators randomly. Similarly, `UnknownHostException` is caught by two validators and is affected by the same kind of issue.

I think the validation of the URL should be deterministic and the API users should never receive misleading error messages.

Reply:
I have modified the annotations so that more accurate error messages are returned if these exceptions are thrown... I thought of coming back to a "single annotation validation" with these custom error messages too, but I rather wait until I receive feedback from you, since I'm not very sure if what I sent is acceptable or not...

Reply:
All right, I've gone back to the single annotation with multiple error messages version.

Reply:
Sorry for the going back and forth :D

Reply:
Don't worry! It's actually my bad, since I made you all review this a few times... I learned about overriding the default messages, so I think it's been positive! Thanks @josejulio ! :smile: